frontend/app/src/store/history/index.ts

```diff
@@ -1,17 +1,916 @@
+      }
+
+      if (firstLoad || refresh) {
+        fetchAssociatedLocations().then();
+      }
+
+      const fetchOnlyCache = async () => {
+        ledgerActions.value = await fetchLedgerActionsHandler(true);
+      };
+
+      setStatus(firstLoad ? Status.LOADING : Status.REFRESHING);
+
+      if (!onlyLocation) await fetchOnlyCache();
+
+      if (!onlyCache || onlyLocation) {
+        setStatus(Status.REFRESHING);
+        const { notify } = useNotifications();
+
+        const exchange = onlyLocation
+          ? exchangeName(onlyLocation as TradeLocation)
+          : i18n.tc('actions.ledger_actions.all_exchanges');
+
+        await fetchLedgerActionsHandler(false, {
+          location: onlyLocation
+        }).catch(error => {
+          notify({
+            title: i18n.tc('actions.ledger_actions.error.title', undefined, {
+              exchange
+            }),
+            message: i18n.tc(
+              'actions.ledger_actions.error.description',
+              undefined,
+              {
+                exchange,
+                error
+              }
+            ),
+            display: true
+          });
+        });
+
+        if (!onlyLocation) await fetchOnlyCache();
+      }
+
+      setStatus(
+        isTaskRunning(taskType).value ? Status.REFRESHING : Status.LOADED
+      );
+    } catch (e) {
+      logger.error(e);
+      setStatus(Status.NONE);
+    }
+  };
+
+  const updateLedgerActionsPayload = (
+    newPayload: Partial<LedgerActionRequestPayload>
+  ) => {
+    if (!isEqual(ledgerActionsPayload.value, newPayload)) {
+      ledgerActionsPayload.value = newPayload;
+      fetchLedgerActions().then();
+    }
+  };
+
+  const addLedgerAction = async (
+    ledgerAction: NewLedgerAction
+  ): Promise<ActionStatus> => {
+    let success = false;
+    let message = '';
+    try {
+      await api.history.addLedgerAction(ledgerAction);
+      success = true;
+    } catch (e: any) {
+      message = e.message;
+    }
+
+    fetchAssociatedLocations().then();
+    fetchLedgerActions().then();
+    return { success, message };
+  };
+
+  const editLedgerAction = async (
+    ledgerAction: LedgerActionEntry
+  ): Promise<ActionStatus> => {
+    let success = false;
+    let message = '';
+    try {
+      await api.history.editLedgerAction(ledgerAction);
+      success = true;
+    } catch (e: any) {
+      message = e.message;
+    }
+
+    fetchAssociatedLocations().then();
+    fetchLedgerActions().then();
+    return { success, message };
+  };
+
+  const deleteLedgerAction = async (
+    identifier: number
+  ): Promise<ActionStatus> => {
+    let success = false;
+    let message = '';
+    try {
+      success = await api.history.deleteLedgerAction(identifier);
+    } catch (e: any) {
+      message = e.message;
+    }
+
+    fetchAssociatedLocations().then();
+    fetchLedgerActions().then();
+    return { success, message };
+  };
+
+  // Ignored
+  const ignored = ref({}) as Ref<IgnoredActions>;
```

Comment:
woudn't `ref<IgnoredActions>({})` work here with type inference instead of casting?

Reply:
I just remembered casting is only needed due to a bug that strips some types of properties that affect the inference for `BigNumber` if you don't hit this bug the is no need to cast.

Reply:
alright thanks for the explanation. I did this because it show error but I don't know why, so I just create make the state using this kind of writing.